File: elixir/test/mandator_clearing_export_download_entity_test.exs
# MandatorClearingExportDownload entity test (offline, mock transport)

defmodule BluefinTecsMerchantServices.MandatorClearingExportDownloadEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsMerchantServices.Helpers, as: H
  alias BluefinTecsMerchantServices.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/mandator_clearing_export_download/MandatorClearingExportDownloadTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsMerchantServices.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.mandator_clearing_export_download"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsMerchantServices.test()
    ent = BluefinTecsMerchantServices.mandator_clearing_export_download(sdk)
    assert ent != nil
  end

  test "should load an existing record" do
    id = first_id()

    if id != nil do
      sdk = mk_sdk()
      ent = BluefinTecsMerchantServices.mandator_clearing_export_download(sdk)
      rec = BluefinTecsMerchantServices.Entity.MandatorClearingExportDownload.load(ent, S.jm(["id", id]))
      assert S.ismap(rec)
      assert S.getprop(rec, "id") == id
    end
  end

  test "should create then read back" do
    sdk = BluefinTecsMerchantServices.test(S.jm(["entity", S.jm(["mandator_clearing_export_download", S.jm([])])]))
    ent = BluefinTecsMerchantServices.mandator_clearing_export_download(sdk)
    made = BluefinTecsMerchantServices.Entity.MandatorClearingExportDownload.create(ent, S.jm(["name", "test-create"]))
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end

File: scala/sdktest/MandatorClearingExportDownloadEntityTest.scala
// Generated basic-flow test for the mandator_clearing_export_download entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped MandatorClearingExportDownloadTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantservicessdk.core.{Helpers, SdkEntity, BluefinTecsMerchantServicesSDK}
import voxgig.bluefintecsmerchantservicessdk.utility.struct.Struct

object MandatorClearingExportDownloadEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("mandator_clearing_export_download.instance") {
      val testsdk = BluefinTecsMerchantServicesSDK.testSDK()
      val ent = testsdk.mandatorClearingExportDownload(null)
      rep.check("mandator_clearing_export_download.instance", ent != null, "expected non-null mandator_clearing_export_download entity")
    }

    rep.scope("mandator_clearing_export_download.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/mandator_clearing_export_download/MandatorClearingExportDownloadTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantServicesSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("mandator_clearing_export_download01", "MANDATOR_CLEARING_EXPORT_DOWNLOAD01")
      idmap.put("mandator_clearing_export_download02", "MANDATOR_CLEARING_EXPORT_DOWNLOAD02")
      idmap.put("mandator_clearing_export_download03", "MANDATOR_CLEARING_EXPORT_DOWNLOAD03")
      val now = System.currentTimeMillis()

      // CREATE
      val mandatorClearingExportDownloadRef01Ent = client.mandatorClearingExportDownload(null)
      var mandatorClearingExportDownloadRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.mandator_clearing_export_download"), "mandator_clearing_export_download_ref01"))
      val mandatorClearingExportDownloadRef01DataResult = mandatorClearingExportDownloadRef01Ent.create(mandatorClearingExportDownloadRef01Data, null)
      mandatorClearingExportDownloadRef01Data = Helpers.toMapAny(mandatorClearingExportDownloadRef01DataResult match { case e: SdkEntity => e.data(); case o => o })
      rep.check("mandator_clearing_export_download.create.map", mandatorClearingExportDownloadRef01Data != null, "expected create result to be a map")
      rep.check("mandator_clearing_export_download.create.id", mandatorClearingExportDownloadRef01Data != null && mandatorClearingExportDownloadRef01Data.get("id") != null, "expected created entity to have an id")

      // LOAD
      val mandatorClearingExportDownloadRef01MatchDt0 = new LinkedHashMap[String, Object]()
      mandatorClearingExportDownloadRef01MatchDt0.put("id", mandatorClearingExportDownloadRef01Data.get("id"))
      val mandatorClearingExportDownloadRef01DataDt0Loaded = mandatorClearingExportDownloadRef01Ent.load(mandatorClearingExportDownloadRef01MatchDt0, null)
      val mandatorClearingExportDownloadRef01DataDt0LoadResult = Helpers.toMapAny(mandatorClearingExportDownloadRef01DataDt0Loaded match { case e: SdkEntity => e.data(); case o => o })
      rep.check("mandator_clearing_export_download.load.map", mandatorClearingExportDownloadRef01DataDt0LoadResult != null, "expected load result to be a map")
      rep.eq("mandator_clearing_export_download.load.id", mandatorClearingExportDownloadRef01Data.get("id"), mandatorClearingExportDownloadRef01DataDt0LoadResult.get("id"))
    }
  }
}
